code/obj/item/storage/backpack_belt_etc.dm

```diff
@@ -293,45 +289,40 @@
 	name = "medical belt"
 	icon_state = "injectorbelt"
 	item_state = "injector"
-	can_hold = list(
-		/obj/item/robodefibrillator
-	)
-	in_list_or_max = 1
+
+	New()
+		..()
+		AddComponent(/datum/component/storage, can_hold = list(/obj/item/robodefibrillator), in_list_or_max = TRUE, max_wclass = 2, does_not_open_in_pocket = FALSE)
 
 /obj/item/storage/belt/mining
 	name = "miner's belt"
 	desc = "Can hold various mining tools."
 	icon_state = "minerbelt"
 	item_state = "utility"
-	can_hold = list(
-		/obj/item/mining_tool,
-		/obj/item/mining_tools
-	)
-	in_list_or_max = 1
+
+	New()
+		..()
+		AddComponent(/datum/component/storage, can_hold = list(/obj/item/mining_tool, /obj/item/mining_tools), in_list_or_max = TRUE, max_wclass = 2, does_not_open_in_pocket = FALSE)
 
 /obj/item/storage/belt/hunter
 	name = "trophy belt"
 	desc = "Holds normal-sized items, such as skulls."
 	icon_state = "minerbelt"
 	item_state = "utility"
-	max_wclass = 3
+
+	New()
+		..()
+		AddComponent(/datum/component/storage, max_wclass = 3, does_not_open_in_pocket = FALSE)
 
 /obj/item/storage/belt/security
 	name = "security toolbelt"
 	desc = "For the trend-setting officer on the go. Has a place on it to clip a baton and a holster for a small gun."
 	icon_state = "secbelt"
 	item_state = "secbelt"
-	can_hold = list(/obj/item/baton, // not included in this list are guns that are already small enough to fit (like the detective's gun)
-	/obj/item/gun/energy/taser_gun,
-	/obj/item/gun/energy/phaser_gun,
-	/obj/item/gun/energy/laser_gun,
-	/obj/item/gun/energy/egun,
-	/obj/item/gun/energy/lawbringer,
-	/obj/item/gun/energy/lawbringer/old,
-	/obj/item/gun/energy/wavegun,
-	/obj/item/gun/kinetic/revolver,
-	/obj/item/gun/kinetic/zipgun)
-	in_list_or_max = 1
+
+	New()
+		..()
+		AddComponent(/datum/component/storage, can_hold = list(/obj/item/baton, /obj/item/gun), in_list_or_max = TRUE, max_wclass = 2, does_not_open_in_pocket = FALSE)
```

Comment:
The old `can_hold` list was kinda redundant, because you already can't hold all the super dangerous guns cuz they're too big to fit.

Reply:
Doesn't being in the list *bypass* the size restriction?

Reply:
`var/in_list_or_max = 0 // sorry for the dumb var name - if can_hold has stuff in it, if this is set, something will fit if it's at or below max_wclass OR if it's in can_hold, otherwise only things in can_hold will fit`
The modified behaviour here will let you put *any* gun into the belt, doesn't seem like a good idea nor an intended change.